src/contract.rs

```diff
@@ -0,0 +1,314 @@
+        let current_transaction_id = self.get_transaction_id(transaction_id);
+
+        let wallet = get_mut_wallet(&mut self.wallets, wallet_id);
+        check_seller(wallet.seller);
+        assert_eq!(wallet.state, WalletState::AwaitingConfirmation);
+
+        if transfer_tokens(
+            current_transaction_id,
+            &self.ft_program_id,
+            &exec::program_id(),
+            &wallet.buyer,
+            wallet.amount,
+        )
+        .await
+        .is_ok()
+        {
+            wallet.state = WalletState::AwaitingDeposit;
+
+            self.transactions.remove(&current_transaction_id);
+
+            reply(EscrowEvent::Refunded(current_transaction_id, wallet_id));
+        } else {
+            reply(EscrowEvent::TransactionFailed);
+        }
+    }
+
+    async fn cancel(&mut self, wallet_id: WalletId) {
+        let wallet = get_mut_wallet(&mut self.wallets, wallet_id);
+        check_buyer_or_seller(wallet.buyer, wallet.seller);
+        assert_eq!(wallet.state, WalletState::AwaitingDeposit);
+
+        wallet.state = WalletState::Closed;
+
+        reply(EscrowEvent::Cancelled(wallet_id));
+    }
+
+    /// Continues cached transaction by `transaction_id`.
+    ///
+    /// Execution makes sense if, when returning from an async message,
+    /// the gas ran out and the state has changed.
+    async fn continue_transaction(&mut self, transaction_id: u64) {
+        let transactions = self.transactions.clone();
+        let payload = &transactions
+            .get(&transaction_id)
+            .expect("Transaction does not exist");
+        if let Some(action) = payload {
+            match action {
+                EscrowAction::Deposit(wallet_id) => {
+                    self.deposit(Some(transaction_id), *wallet_id).await
+                }
+                EscrowAction::Confirm(wallet_id) => {
+                    self.confirm(Some(transaction_id), *wallet_id).await
+                }
+                EscrowAction::Refund(wallet_id) => {
+                    self.refund(Some(transaction_id), *wallet_id).await
+                }
+                _ => unreachable!(),
+            }
+        } else {
+            msg::reply(EscrowEvent::TransactionProcessed, 0)
+                .expect("Error in a reply `EscrowEvent::TransactionProcessed`");
+        }
+    }
+
+    fn get_transaction_id(&mut self, transaction_id: Option<u64>) -> u64 {
+        match transaction_id {
+            Some(transaction_id) => transaction_id,
+            None => {
+                let transaction_id = self.transaction_id;
+                self.transaction_id = self.transaction_id.wrapping_add(1);
+                transaction_id
+            }
+        }
+    }
+}
+
+static mut ESCROW: Option<Escrow> = None;
+
+#[no_mangle]
+extern "C" fn init() {
+    let config: InitEscrow = msg::load().expect("Unable to decode InitEscrow");
+
+    if config.ft_program_id == ActorId::zero() {
```

Comment:
We have [`ActorId::is_zero`](https://docs.gear.rs/gstd/struct.ActorId.html#method.is_zero) function for such purposes.